### PR TITLE
test: migrate `cargo_alias_config&cargo_config/mod` to snapbox

### PR DESCRIPTION
**Pull request by heisen-li**
### What does this PR try to resolve?

Resolve the following file migration:
`tests/testsuite/cargo_alias_config.rs`
`tests/testsuite/cargo_config/mod.rs`

part of #14039

### CODE REVIEW

File: tests/testsuite/cargo_config/mod.rs [hunk 1]
```diff
@@ -81,168 +82,149 @@ fn get_toml() {
...
         .env("CARGO_ALIAS_BAR", "cat dog")
         .env("CARGO_BUILD_JOBS", "100")
-        .with_json(all_json)
-        .with_stderr(
-            "\
-note: The following environment variables may affect the loaded values.
+        .with_stdout_data(r#"
+{"alias":{"foo":"abc --xyz","sub-example":["sub","example"]},"build":{"jobs":99,"rustflags":["--flag-global","--flag-directory"]},"extra-table":{"somekey":"somevalue"},"profile":{"dev":{"opt-level":3,"package":{"foo":{"opt-level":1}}}},"target":{"cfg(target_os = \"linux\")":{"runner":"runme"}}}
+
+"#.json_lines())
```
Comment (weihanglo) [inline tests/testsuite/cargo_config/mod.rs:185]: `.json()` is sufficient. JSON lines are mainly for `--message-format=json` like output. Also, within a single JSON object we are supposed to be able to pretty print it. Pretty-printed JSON objects are easier to review.


```suggestion
{
  "alias": {
    "foo": "abc --xyz",
    "sub-example": [
      "sub",
      "example"
    ]
  },
  "build": {
    "jobs": 99,
    "rustflags": [
      "--flag-global",
      "--flag-directory"
    ]
  },
  "extra-table": {
    "somekey": "somevalue"
  },
  "profile": {
    "dev": {
      "opt-level": 3,
      "package": {
        "foo": {
          "opt-level": 1
        }
      }
    }
  },
  "target": {
    "cfg(target_os = \"linux\")": {
      "runner": "runme"
    }
  }
}

"#.json())
```

File: tests/testsuite/cargo_config/mod.rs [hunk 2]
```diff
@@ -81,168 +82,149 @@ fn get_toml() {
...
     // json-value is the same for the entire root table
     cargo_process("config get --format=json-value -Zunstable-options")
         .cwd(&sub_folder.parent().unwrap())
         .masquerade_as_nightly_cargo(&["cargo-config"])
-        .with_json(all_json)
-        .with_stderr(
-            "\
-note: The following environment variables may affect the loaded values.
+        .with_stdout_data(r#"
+{"alias":{"foo":"abc --xyz","sub-example":["sub","example"]},"build":{"jobs":99,"rustflags":["--flag-global","--flag-directory"]},"extra-table":{"somekey":"somevalue"},"profile":{"dev":{"opt-level":3,"package":{"foo":{"opt-level":1}}}},"target":{"cfg(target_os = \"linux\")":{"runner":"runme"}}}
```
Comment (weihanglo) [inline tests/testsuite/cargo_config/mod.rs:200]: ditto for single json and pretty printing.

File: tests/testsuite/cargo_config/mod.rs (hunk 1 above)
Comment (heisen-li) [reply tests/testsuite/cargo_config/mod.rs]: Thanks for the detailed explanation. Modified.

**Comment by weihanglo** [comment]
Thanks.

@bors r+

**Pull request merged**